src/UIAutomation/FunctionalTests/UiaOperationAbstractionTests.cpp

```diff
@@ -11,6 +11,79 @@
 
 using namespace UiaOperationAbstraction;
 
+// Test framework extensions used for comparing and printing different types via the
+// `Assert` test functions.
+namespace Microsoft::VisualStudio::CppUnitTestFramework
+{
+#pragma warning(push)
+#pragma warning(disable: 4505)
```

Comment:
I'm confused as to why this is needed and if it is needed how this even works at all then.

This warning should be saying that the compiler didn't even generate the code for this function, as it considered it dead. Suppressing the warning shouldn't change that fact (just remove the warning).

Then it would stand to reason that either the compiler did generate the code (and the warning is spurious) or the compile didn't generate the code (and suppressing the warning is just silencing a real issue).

Since this probably works, the only other explanation is that I'm misunderstanding something -- could you help me connect the dots? :)

Reply:
To be completely honest... I copied the existing code to just quickly draft some test changes and completely forgot about getting back to it. So... I am not sure but I also cannot see any warnings for these specializations, so I removed them.

(thanks for catching it!)

Reply:
So, the reason this is in TestUtils.h is because `ToString` isn't used in every compilation unit that header is included in. But yeah, agreed it isn't needed in a .cpp file.